### ISSUE TITLE
Show new transactions instantly in activity

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#1715]
Currently when the user submits a transaction, there is a lag time between submission and appearing in the activity tab of the home screen. This lag creates uncertainty for the user as to whether the transaction has actually been submitted successfully.

A potential solution via caching of transactions locally is discussed in https://github.com/blockstack/stacks-wallet-web/discussions/1712

This solution would not only provide greater "speediness" but also prevent users from thinking their transactions have failed to go through at all, should the API experience issues with recognizing transactions per https://github.com/blockstack/stacks-blockchain-api/issues/765 / https://github.com/hirosystems/devops/issues/797.

Companion issue for the desktop wallet: https://github.com/blockstack/stacks-wallet/issues/921

**Comment by Eshwari007** [comment]
Note:All the below occurred in one same account in Mainnet

I have noticed the similar behavior and was even thinking of logging the issue but after couple minutes,I noticed the missing transaction in the account
There were two transaction but the first one was not visible in Activity despite tx was successful.
Within few sec,I placed another tx but that received nonce conflict.
I then tried again in few min on another tx and the tx was successful
This time I was able to view the latest tx as pending.
I even went to explorer to check on the history of the account but was not able to track the first tx.

After few minutes once the latest tx was completed,and the balance was deducted.I saw both of the tx (first and last) in the activity and also in explorer.

There were only Failed to load resources error in the console.
<img width="1254" alt="tx error" src="https://user-images.githubusercontent.com/38872890/133653246-97d89380-ea07-4715-a504-0bd20c476274.png">
